docs/debugging/cluster-reboot/main.go

```diff
@@ -0,0 +1,511 @@
+	fmt.Println(" hostfile   Generates hostfiles in `-folder`")
+	fmt.Println(" reboot     Reboots servers defined in `-hostfile`")
+	fmt.Println(" monitor    Monitors the uptime of hosts defined in `-hostfile`")
+	fmt.Println(" -----------------------------")
+	fmt.Println("")
+}
+
+func makeClient() (err error) {
+	mclient, err = madmin.NewWithOptions(endpoint, &madmin.Options{
+		Creds:     credentials.NewStaticV4(miniokey, miniosecret, ""),
+		Secure:    secure,
+		Transport: DefaultTransport(secure),
+	})
+	return
+}
+
+func sets() {
+	pools, _, err := getInfra()
+	if err != nil {
+		panic(err)
+	}
+	sets := make(map[int]map[int][]string)
+	for pid, p := range pools {
+		sets[pid] = make(map[int][]string, 0)
+		for _, s := range p.Servers {
+			for _, set := range s.Sets {
+				sets[pid][set.ID] = append(sets[pid][set.ID], s.Endpoint)
+			}
+		}
+	}
+	if jsonOutput {
+		jsonOut(sets)
+		return
+	}
+	for i, v := range sets {
+		fmt.Println("--------- POOL:", i, "----------")
+		for ii, vv := range v {
+			fmt.Println("--------- SET:", ii, "-----------")
+			for _, vvv := range vv {
+				fmt.Println(vvv)
+			}
+		}
+	}
+}
+
+func getInfra() (pools map[int]*Pool, totalServers int, err error) {
+	makeClient()
+	var info madmin.InfoMessage
+	info, err = mclient.ServerInfo(context.Background())
+	if err != nil {
+		return
+	}
+
+	// bb, err := os.ReadFile("infra.json")
+	// if err != nil {
+	// 	panic(err)
+	// }
+	//
+	// err = json.Unmarshal(bb, &info)
+	// if err != nil {
+	// 	panic(err)
+	// }
+	//
+	pools = make(map[int]*Pool, 0)
+	for i := range info.Pools {
+		pools[i+1] = &Pool{
+			Servers: make(map[string]*Server, 0),
+		}
+	}
+
+	totalServers = 0
+	for _, s := range info.Servers {
+		if s.State == "offline" {
+			continue
+		}
+
+		totalServers++
+		pool := pools[s.PoolNumber]
+		server, ok := pool.Servers[s.Endpoint]
+		if !ok {
+			pool.Servers[s.Endpoint] = &Server{
+				Sets:     make(map[int]*Set, 0),
+				Rebooted: false,
+				Endpoint: s.Endpoint,
+			}
+			server = pool.Servers[s.Endpoint]
+		}
+
+		for _, d := range s.Disks {
+			set, ok := server.Sets[d.SetIndex]
+			if !ok {
+				server.Sets[d.SetIndex] = &Set{
+					Disks:      make(map[string]*Disk, 0),
+					SCParity:   info.Backend.StandardSCParity,
+					RRSCParity: info.Backend.RRSCParity,
+					ID:         d.SetIndex,
+					Pool:       d.PoolIndex,
+					CanReboot:  false,
+				}
+				set = server.Sets[d.SetIndex]
+			}
+
+			set.Disks[d.Endpoint] = &Disk{
+				UUID:           d.UUID,
+				Index:          d.DiskIndex,
+				Pool:           d.PoolIndex,
+				Server:         d.Endpoint,
+				Set:            d.SetIndex,
+				Path:           d.DrivePath,
+				State:          d.State,
+				UsedPercentage: (math.Ceil((float64(d.UsedSpace)/float64(d.TotalSpace)*100)*100) / 100),
+			}
+		}
+	}
+
+	return
+}
+
+func makeHostfile() {
+	pools, totalServers, err := getInfra()
+	var rebootRounds [200][200]map[string]*Server
```

Comment:
```suggestion
	const maxRounds = 200
	const maxPools = 200
	var rebootRounds [maxRounds][maxPools]map[string]*Server
```

That is correct, right?

Reply:
indeed